WireFoundation/Sources/WireFoundation/Crypto/AES256Crypto.swift

```diff
@@ -0,0 +1,203 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CommonCrypto
+import Foundation
+
+/// A namespace for AES256 cryptographic operations.
+
+public enum AES256Crypto {
+
+    // MARK: - Keys
+
+    /// Generate a random encryption key for AES256 cryptography.
+    /// - Returns: A 32-byte key.
+
+    public static func generateRandomEncryptionKey() throws -> Data {
+        try SecureRandomByteGenerator.generateBytes(count: UInt(kCCKeySizeAES256))
+    }
+
+    // MARK: - Prefixed IV
+
+    /// Encrypt data with a prefixed IV all at once.
+    ///
+    /// This method will prefix a random IV before encrypting. To decrypt back
+    /// into plaintext, you must use the corresponding `decryptAllAtOnceWithPrefixedIV`
+    /// function.
+    ///
+    /// The entire cryptographic operation will occur in memory, therefore
+    /// this method should only be used with relatively small data.
+    ///
+    /// - Parameters:
+    ///   - plaintext: The data to encrypt.
+    ///   - key: The encryption key. It must be 32 bytes.
+    ///
+    /// - Returns: The encrypted ciphertext.
+
+    public static func encryptAllAtOnceWithPrefixedIV(
+        plaintext: Data,
+        key: Data
+    ) throws -> Data {
```

Comment:
would be a good idea to wrap this data into a custom type? as you don't know this data needs to be decrypted by `decryptAllAtOnceWithPrefixedIV`

Reply:
done 6cf6efff5cd173ff242566c15629660d4ef1ffe8